modules/search/client/components/SearchUsers.component.js

```diff
@@ -0,0 +1,182 @@
+        <div className="col-xs-12" ng-class="{'col-sm-8': ContactsList.contacts.length >= 6, 'text-center': ContactsList.contacts.length < 6}">
+          <h4 className="text-muted">
+
+            {users.length === 1 &&
+            <span>
+            One user found.
+            </span>
+            }
+
+            {users.length > 1 &&
+            <span>
+              {users.length} users found.
+              {/* {{ (ContactsList.contacts | filter:{ confirmed: true }).length }} contacts */}
+            </span>
+            }
+
+            <UsersList users={users}/>
+
+          </h4>
+        </div>
+      </div>
+    </div>
+  );
+
+  const noUsers = (
+    <div className="row content-empty" ng-if="ContactsList.contacts.$resolved && !ContactsList.contacts.length">
+      <i className="icon-3x icon-users"></i>
+      <h4>No users found with this name.</h4>
+    </div>
+  );
+
+  return (
+    <div>
+      {!resolved?
+        <div className="content-wait"
+          role="alertdialog"
+          aria-busy="true"
+          aria-live="assertive"
+          ng-if="!ContactsList.contacts || !ContactsList.contacts.$resolved">
+          <small>Wait a moment...</small>
+        </div>
+        :
+        (users && users.length > 0? userList : noUsers)
+      }
+    </div>
+  );
+}
+
+UsersResults.propTypes = {
+  users: PropTypes.array,
+  resolved: PropTypes.bool
+};
+
+// export class SearchUsersBar extends React.Component {
+
+// export default function SearchUsers({ searchQuery }) {
+class SearchUsers extends React.Component {
+
+  constructor(props) {
+    super(props);
+    console.log('SearchUsers constructor');
+    this.handleChange = this.handleChange.bind(this);
+    this.actionSearch = this.actionSearch.bind(this);
+
+    this.state = {
+      searchQuery: props.searchQuery || '',
+      resolved: true
+    };
+  }
+
+  handleChange(event) {
+    this.setState({ searchQuery: event.target.value });
+    // this.props.searchQuery = event.target.value ;
+  }
+
+  async actionSearch(event) {
+    event.preventDefault();
+    event.stopPropagation();
+    this.setState({ resolved: false });
+    console.log('Searching for: ' + this.state.searchQuery);
+
+    // eslint-disable-next-line react/no-access-state-in-setstate
+    const userResults = await searchUsers(this.state.searchQuery);
+    console.log(userResults);
+    this.setState({ resolved: true, users: userResults.data });
+    console.log(this.state.users);
+  };
+
+  clickClear() {
+    this.setState({ searchQuery: '' });
+  }
+
+  render() {
+    console.log('SearchUsers: render');
+
+    const borderStyle = { 'borderLeftWidth': '1px',
+      'borderLeftStyle': 'solid',
+      'borderColor': '#eee' };
+
+    const switchToSearchPlaces = (
+      <span className="input-group-btn"
+        style={borderStyle}>
+        <a type="button"
+          href="/search"
+          className="btn btn-lg btn-default"
+          tooltip-enable="::search.screenWidth >= 768"
```

Comment:
A little bit of Angular tooltip directive here that can be removed.

Reply:
Well, I left all the angular stuff just to make sure they were being correctly translated to react (at least, the basic ones), but I can remove them very easily.

Reply:
@simison I'm thinking to better leave the tooltip directives just in case we got something similar working in the future for React? At the end, they don't really annoy so much, do they?

Reply:
It's really better to remove anything that's not used. 🙂 It can be confusing to have dead code around. If we create tooltips in future (probably not as they're not great user interface) it'll look different to Angular directives so they won't work "out of the box" with this syntax anyway.